File: Employee-Management-System/src/main/java/com/example/employeemanagementsystem/Service/EmployeeService.java
package com.example.employeemanagementsystem.Service;

import com.example.employeemanagementsystem.Model.Employee;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public interface EmployeeService {
    Employee saveEmployee(Employee employee);
    Optional<Employee> getEmployeeById(int id);
    List<Employee> getAllEmployee();
    Employee updateEmployee(int id, Employee employee);
    void deleteEmployee(int id);
}

File: Employee-Management-System/employee-management-frontend/src/service/EmployeeService.js
import axios from "axios";

const BASE_URL = "http://localhost:8080/employee";
const GETALL_URL = "http://localhost:8080/employee/all";
const SAVE_URL="http://localhost:8080/employee/add"
const UPDATE_URL = "http://localhost:8080/employee/{id}/update"
const BYID_URL= "http://localhost:8080/employee/{id}";

class EmployeeService {

    getAllEmployee() {
        return axios.get(GETALL_URL)
    }

    saveEmployee(employeeData) {
        return axios.post(SAVE_URL, employeeData);
    }
    updateEmployee(id, employeeData){
        return axios.put(`${BASE_URL}/${id}/update`, employeeData);
    }
    getEmployeeById(id){
        return axios.get(`${BASE_URL}/${id}`);
    }
    deleteEmployee(id){
        return axios.delete(`${BASE_URL}/${id}/delete`);
    }
}

export default new EmployeeService();